### ISSUE TITLE
List CLI command crashes with traceback

### ISSUE THREAD

**Issue by sfayer** [ic-hep/pdm#224]
```
Traceback (most recent call last):
  File "./bin/pdm", line 10, in <module>
    pdmcli.main(os.path.join(TOP_PATH, "etc", "system.client.conf"))
  File "/home/centos/pdm/src/pdm/CLI/pdmcli.py", line 84, in main
    args.func(args)
  File "/home/centos/pdm/src/pdm/CLI/user_subcommand.py", line 216, in list
    self._print_formatted_listing(listing)
  File "/home/centos/pdm/src/pdm/CLI/user_subcommand.py", line 252, in _print_formatted_listing
    size_len = len(str(max(d['size'] for d in listing)))
  File "/home/centos/pdm/src/pdm/CLI/user_subcommand.py", line 252, in <genexpr>
    size_len = len(str(max(d['size'] for d in listing)))
TypeError: string indices must be integers
```

The input listing dictionary is:
```
{u'gsiftp://pdmtest1.grid.hep.ph.ic.ac.uk:49999/etc/default': [{u'st_ctime': 0, u'st_mtime': 1526649952, u'name': u'.', u'st_nlink': 1, u'st_ino': 0, u'st_dev': 0, u'st_size': 44, u'st_gid': 0, u'st_uid': 0, u'st_mode': 33261, u'st_atime': 0}, {u'st_ctime': 0, u'st_mtime': 1526774048, u'name': u'..', u'st_nlink': 1, u'st_ino': 0, u'st_dev': 0, u'st_size': 8192, u'st_gid': 0, u'st_uid': 0, u'st_mode': 33261, u'st_atime': 0}, {u'st_ctime': 0, u'st_mtime': 1526649952, u'name': u'grub', u'st_nlink': 1, u'st_ino': 0, u'st_dev': 0, u'st_size': 302, u'st_gid': 0, u'st_uid': 0, u'st_mode': 33188, u'st_atime': 0}, {u'st_ctime': 0, u'st_mtime': 1523346843, u'name': u'nss', u'st_nlink': 1, u'st_ino': 0, u'st_dev': 0, u'st_size': 1756, u'st_gid': 0, u'st_uid': 0, u'st_mode': 33188, u'st_atime': 0}, {u'st_ctime': 0, u'st_mtime': 1478283873, u'name': u'useradd', u'st_nlink': 1, u'st_ino': 0, u'st_dev': 0, u'st_size': 119, u'st_gid': 0, u'st_uid': 0, u'st_mode': 33188, u'st_atime': 0}]}
```

It looks like the field names and format have changed quite a lot since the last version...

**Comment by martynia** [comment]
Yes, There is a job number added, so the elements are now tuples  (job no, endpoint) in gfal . Alex is probably returning this. Not clear to me why, since the end user does not care about the job number...
J.

**Comment by sfayer** [comment]
Nah, this call isn't receiving any job numbers (or tuples), but the structure and field names have still changed: I don't think it'll be that hard to fix, although you will have to write a function to convert the st_mode field to the human readable "-rwxr-x---" style format. Make sure you implement support for the less common mode values, (like sticky bits) as well.

**Comment by alexanderrichards** [comment]
hi all, This is because I used to parse the listing output in the workqueue using my regex which named all the keys and had the human readable permissions bit. Since @martynia  gfal2-ls tool now does the listing and formats it into a return dictionary we have the keys that come from the stat object. I did consider transmuting them back to what we used to have but since 
 - I would have to essentially loop through and end up with a very similar dictionary just with different key names
 - There are more fields from the stat object that my regex gave
 - It would add another layer of processing = more to go wrong + more to maintain.

I didn't bother
